src/filters/transform/vsfilter/DirectVobSubFilter.cpp

```diff
@@ -2280,10 +2280,19 @@ void CDirectVobSubFilter::SetYuvMatrix()
         }
         else
         {
-            yuv_matrix = (extformat.VideoTransferMatrix == DXVA2_VideoTransferMatrix_BT709 ||
-                          extformat.VideoTransferMatrix == DXVA2_VideoTransferMatrix_SMPTE240M ||
-                          m_xy_size_opt[SIZE_ORIGINAL_VIDEO].cx > m_bt601Width ||
-                          m_xy_size_opt[SIZE_ORIGINAL_VIDEO].cy > m_bt601Height) ? ColorConvTable::BT709 : ColorConvTable::BT601;
+            switch (extformat.VideoTransferMatrix)
+            {
+            case DXVA2_VideoTransferMatrix_BT709:
+                yuv_matrix = ColorConvTable::BT709;
+                break;
+            case DXVA2_VideoTransferMatrix_SMPTE240M:
+                yuv_matrix = ColorConvTable::BT601;

```

Comment:
SMPTE 240M and BT.601 have different matrices.
This differs from the original code, and neither seems correct.

See: https://learn.microsoft.com/en-us/windows/win32/api/dxva2api/ne-dxva2api-dxva2_videotransfermatrix

Reply:
Thanks! I probably mixed up SMPTE240M and Bt601 while writing. Should now be fixed, though actual SMPTE240 isn't supported by ColorConvTable (yet) so it gets conflated with BT709 again (for now) like in the original code.